src/test/ui/issues/issue-76191.stderr

```diff
@@ -0,0 +1,22 @@
+error[E0308]: mismatched types
+  --> $DIR/issue-76191.rs:10:9
+   |
+LL | const RANGE: RangeInclusive<i32> = 0..=255;
+   | ------------------------------------------- constant defined here
+...
+LL |     match n {
+   |           - this expression has type `i32`
+LL |         RANGE => {}
+   |         ^^^^^
+   |         |
+   |         expected `i32`, found struct `std::ops::RangeInclusive`
+   |         `RANGE` is interpreted as a constant, not a new binding
+   |         help: introduce a new binding instead: `other_range`
+   |
+   = note: expected type `i32`
+            found struct `std::ops::RangeInclusive<i32>`
+   = note: Constants only support matching the same type. If you are using range syntax (i.e. ..= or similar) in your constant you may want to move the range into the match block
```

Comment:
Ideally the output would look closer to the following:

```
error[E0308]: mismatched types
  --> $DIR/issue-76191.rs:10:9
   |
LL | const RANGE: RangeInclusive<i32> = 0..=255;
   | ------------------------------------------- constant defined here
...
LL |     match n {
   |           - this expression has type `i32`
LL |         RANGE => {}
   |         ^^^^^
   |         |
   |         expected `i32`, found struct `std::ops::RangeInclusive`
   |         `RANGE` is interpreted as a constant, not a new binding
   |
   = note: expected type `i32`
            found struct `std::ops::RangeInclusive<i32>`
   = note: constants only support matching the same type
help: you may want to move the range into the match block
   |
LL |         0..=255 => {}
   |         ^^^^^^^
```
The later suggestion might not be feasible (depending on whether you can get the `hir::Expr` corresponding to the `const`).

Reply:
(This is not a blocker).

Reply:
Oh hmm, I can look into doing that, but would likely make that a separate pull request